src/aap_eda/services/project/scm.py

```diff
@@ -0,0 +1,280 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import logging
+import os
+import shutil
+import tempfile
+from typing import Iterable, Optional
```

Comment:
deprecated import for iterable. Use `from collections.abc import Iterable` instead. 
Refs:
https://docs.astral.sh/ruff/rules/deprecated-import/
https://docs.python.org/3.11/library/typing.html